saleor/graphql/product/bulk_mutations/products.py

```diff
@@ -693,67 +694,40 @@ class Meta:
     @classmethod
     def perform_mutation(cls, root, info, **data):
         errors = defaultdict(list)
-        stocks_input = data["stocks"]
+        stocks = data["stocks"]
         variant = cls.get_node_or_error(
             info, data["variant_id"], only_type=ProductVariant
         )
-        if stocks_input:
-            warehouse_ids = [stock["warehouse"] for stock in stocks_input]
+        if stocks:
+            warehouse_ids = [stock["warehouse"] for stock in stocks]
             cls.check_for_duplicates(warehouse_ids, errors)
             if errors:
                 raise ValidationError(errors)
             warehouses = cls.get_nodes_or_error(
                 warehouse_ids, "warehouse", only_type=Warehouse
             )
-            cls.update_or_create_variant_stocks(variant, stocks_input, warehouses, info)
+            cls.update_or_create_variant_stocks(variant, stocks, warehouses)
 
         variant = ChannelContext(node=variant, channel_slug=None)
         return cls(product_variant=variant)
 
     @classmethod
     @traced_atomic_transaction()
-    def update_or_create_variant_stocks(cls, variant, stocks_input, warehouses, info):
-        stocks_data = []
-        for stock_data, warehouse in zip(stocks_input, warehouses):
-            stock, created = warehouse_models.Stock.objects.get_or_create(
+    def update_or_create_variant_stocks(cls, variant, stocks_data, warehouses):
+        product_variant_stock_trigger = ProductVariantStockWebhookTrigger()
```

Comment:
What does this class do for you? I don't see any prons that you achieved by converting logic to class.  You can do that the same by building standard functions.  If you want to have a separation between logic and API you can always add a file as we have for orders (actions.py)

Reply:
1.The only reason why I've decided to use a class was to store somewhere all stocks with necessary data.  That class is nothing more than a simple container, with two public methods.
2. I have to apply my logic in many different places in our code. In that way, I can inject my logic everywhere where I need it within 4 -5 lines instead of 20-25.

Reply:
We don't use classes in Saleor if we don't really need them. 
You can easily cover your case with two simple function, something like this:  
```

def _is_variant_back_to_stock(updated_stock):
    return (
        not updated_stock.is_created
        and updated_stock.prev_quantity <= 0
        and updated_stock.current_quantity > 0
    )

def trigger_product_variant_back_in_stock_webhook(stocks_data, manager):
    for data in stocks_data:
        if _is_variant_back_to_stock(data):
           manager.product_variant_back_in_stock(
                data.stock.product_variant
            )

def trigger_product_variant_out_of_stock_webhook(stocks_data, manager):
    for data in stocks_data:
        if data.current_quantity <= 0:
            manager.product_variant_out_of_stock(
                data.stock.product_variant
            )
```